src/scf/mixing.jl

```diff
@@ -6,22 +6,30 @@
 
 # The interface is `mix(m, basis, ρin, ρout) -> ρnext`
 
+using LinearMaps
+using IterativeSolvers
+
+@doc raw"""
+Kerker mixing: ``J^{-1} ≈ \frac{α G^2}{(k_F^2 + G^2}``
+where ``k_F`` is the Thomas-Fermi screening constant.
+
+Notes:
+  - Abinit calls ``1/k_F`` the dielectric screening length (parameter *dielng*)
 """
-Kerker mixing: J^-1 ≈ α*G^2/(G0^2 + G^2)
-"""
-struct KerkerMixing{T <: Real}
-    α::T
-    G0::T
+struct KerkerMixing
+    α
+    kF
 end
-# Default parameters suggested by Kresse, Furthmüller 1996 (α=0.8, G0=1.5 Ǎ^{-1})
+# Default parameters suggested by Kresse, Furthmüller 1996 (α=0.8, kF=1.5 Ǎ^{-1})
 # DOI 10.1103/PhysRevB.54.11169
-KerkerMixing() = KerkerMixing(0.8, 0.8)
-function mix(m::KerkerMixing, basis, ρin::RealFourierArray, ρout::RealFourierArray)
+KerkerMixing(;α=0.8, kF=0.8) = KerkerMixing(α, kF)
+function mix(params::KerkerMixing; basis, ρin::RealFourierArray, ρout::RealFourierArray, kwargs...)
```

Comment:
I did it like this because then I can just use the info struct. Otherwise I have to build a slim one first and update it later, which is more coding effort, but it make some things more explicit. I can also do `mix(mixing, basis, ρin, ρout; kwargs...)` and then essentially pass `ρin` twice.

Reply:
I like the second option. I really like passing basically your whole context as kwargs for this kind of API: this way the callee doesn't have to specify what it needs, it can just pick and choose, and the caller can also just fire away all it has at its disposal. However basis, rhoin and rhoout are important parts of what the function do and should be positional.

Reply:
I agree with that philosophy.